python/cudf/cudf/_lib/cpp/types.pxd

```diff
@@ -102,3 +103,5 @@ cdef extern from "cudf/types.hpp" namespace "cudf" nogil:
         HIGHER "cudf::interpolation::HIGHER"
         MIDPOINT "cudf::interpolation::MIDPOINT"
         NEAREST "cudf::interpolation::NEAREST"
+
+    ctypedef int int128 "__int128_t"
```

Comment:
Can we add a comment on what this is doing? I think the only time I've seen this syntax is in some obscure StackOverflow post, and I still don't 100% know what the implications of a Cython alias like this are.

However, I'm also not entirely sure that we need this alias. The only place we use this type is in decimals.pxd where we alias it again to `decimal128`, right? So couldn't we just `ctypedef int __int128_t` here, import it directly with that name in `decimals.pxd`, alias it to `decimal128`, and call it a day? Granting that `__int128_t` is a bit ugly, this would be the one instance of it AFAICT.